lib/AST/ClangTypeConverter.h

```diff
@@ -79,8 +79,9 @@ class ClangTypeConverter :
   /// Swift declaration.
   Decl *getSwiftDeclForExportedClangDecl(const clang::Decl *decl) const;
 
-private:
   clang::QualType convert(Type type);
+
+private:
```

Comment:
If we remove this change we'll have to essentially make `getSubstitutedClangTypesForTemplateFunction` a member of `ClangTypeConverter`. And I don't see any reason that this member _shouldn't_ be public. Let me know which is better.

Reply:
Handling the translation/conversion logic is the responsibility of `ClangTypeConverter`, not that of clients. So it makes sense to have `convert` be private, and expose `getSubstitutedClangTypesForTemplateFunction` as a method on `ClangTypeConverter`, instead of having that logic be a part of `ASTContext`'s implementation. You'll notice that `getClangFunctionType` and `getCanonicalClangFunctionType` are behaving similarly.

Reply:
SGTM. I'll do that.

Reply:
I moved the contents of `getSubstitutedClangTypesForTemplateFunction` to `SwiftTypeConverter:: getSubstitutedClangTypesForTemplateFunction` but I had to also keep `ASTContext:: getSubstitutedClangTypesForTemplateFunction` to wrap the other function. Otherwise (AFAICT) there's no way to call `getSubstitutedClangTypesForTemplateFunction`.